src/catalog/src/remote/manager.rs

```diff
@@ -532,12 +532,51 @@ impl CatalogList for RemoteCatalogManager {
 
     /// List all catalogs from metasrv
     fn catalog_names(&self) -> Result<Vec<String>> {
-        Ok(self.catalogs.load().keys().cloned().collect::<Vec<_>>())
+        Ok(self.catalogs.iter().map(|k| k.key().to_string()).collect())
     }
 
     /// Read catalog info of given name from metasrv.
     fn catalog(&self, name: &str) -> Result<Option<CatalogProviderRef>> {
-        Ok(self.catalogs.load().get(name).cloned())
+        let maybe_catalog = self.catalogs.get(name);
+
+        if let Some(catalog) = maybe_catalog {
+            return Ok(Some(catalog.clone()));
+        }
+
+        let backend = self.backend.clone();
+        let catalog_name = name.to_string();
+
+        // TODO(fys): refactor it later
+        let exist = std::thread::spawn(|| {
+            common_runtime::block_on_write(async move {
+                let mut stream = backend.range(CATALOG_KEY_PREFIX.as_bytes());
+
+                while let Some(Ok(catalog)) = stream.next().await {
+                    let catalog_key = String::from_utf8_lossy(&catalog.0);
+
+                    if let Ok(key) = CatalogKey::parse(&catalog_key) {
```

Comment:
It seems we can do a full sync for catalog here. We can cache not only the requested catalog but also all catalogs from meta.